src/Dependencies/PooledObjects/PooledDelegates.cs

```diff
@@ -0,0 +1,155 @@
+        {
+            var obj = DefaultDelegatePool<TPooled>.Instance.Allocate();
+            obj.Argument = argument;
+            obj.UnboundDelegate = unboundDelegate;
+            boundDelegate = obj.BoundDelegate;
+
+            return new Releaser(obj, obj => DefaultDelegatePool<TPooled>.Instance.Free((TPooled)obj));
+        }
+
+        public static Releaser GetPooledAction<TArg>(Action<TArg> unboundAction, TArg argument, out Action boundAction)
+            => GetPooledDelegate<ActionWithBoundArgument<TArg>, TArg, Action<TArg>, Action>(unboundAction, argument, out boundAction);
+
+        public static Releaser GetPooledAction<T1, TArg>(Action<T1, TArg> unboundAction, TArg argument, out Action<T1> boundAction)
+            => GetPooledDelegate<ActionWithBoundArgument<T1, TArg>, TArg, Action<T1, TArg>, Action<T1>>(unboundAction, argument, out boundAction);
+
+        public static Releaser GetPooledAction<T1, T2, TArg>(Action<T1, T2, TArg> unboundAction, TArg argument, out Action<T1, T2> boundAction)
+            => GetPooledDelegate<ActionWithBoundArgument<T1, T2, TArg>, TArg, Action<T1, T2, TArg>, Action<T1, T2>>(unboundAction, argument, out boundAction);
+
+        public static Releaser GetPooledAction<T1, T2, T3, TArg>(Action<T1, T2, T3, TArg> unboundAction, TArg argument, out Action<T1, T2, T3> boundAction)
+            => GetPooledDelegate<ActionWithBoundArgument<T1, T2, T3, TArg>, TArg, Action<T1, T2, T3, TArg>, Action<T1, T2, T3>>(unboundAction, argument, out boundAction);
+
+        public static Releaser GetPooledFunction<TArg, TResult>(Func<TArg, TResult> unboundFunction, TArg argument, out Func<TResult> boundFunction)
+            => GetPooledDelegate<FuncWithBoundArgument<TArg, TResult>, TArg, Func<TArg, TResult>, Func<TResult>>(unboundFunction, argument, out boundFunction);
+
+        public static Releaser GetPooledFunction<T1, TArg, TResult>(Func<T1, TArg, TResult> unboundFunction, TArg argument, out Func<T1, TResult> boundFunction)
+            => GetPooledDelegate<FuncWithBoundArgument<T1, TArg, TResult>, TArg, Func<T1, TArg, TResult>, Func<T1, TResult>>(unboundFunction, argument, out boundFunction);
+
+        public static Releaser GetPooledFunction<T1, T2, TArg, TResult>(Func<T1, T2, TArg, TResult> unboundFunction, TArg argument, out Func<T1, T2, TResult> boundFunction)
+            => GetPooledDelegate<FuncWithBoundArgument<T1, T2, TArg, TResult>, TArg, Func<T1, T2, TArg, TResult>, Func<T1, T2, TResult>>(unboundFunction, argument, out boundFunction);
+
+        public static Releaser GetPooledFunction<T1, T2, T3, TArg, TResult>(Func<T1, T2, T3, TArg, TResult> unboundFunction, TArg argument, out Func<T1, T2, T3, TResult> boundFunction)
+            => GetPooledDelegate<FuncWithBoundArgument<T1, T2, T3, TArg, TResult>, TArg, Func<T1, T2, T3, TArg, TResult>, Func<T1, T2, T3, TResult>>(unboundFunction, argument, out boundFunction);
+
+        public struct Releaser : IDisposable
```

Comment:
>Releaser [](start = 22, length = 8)

Can we write this without `IClearable`, and perhaps without `_releaser`?
```C#
struct Releaser<TPooled> : IDisposable
   where TPooled : ...
{
    private readonly TPooled _instance;
    internal Releaser(TPooled instance)
    {
        _instance = instance;
    }
    public void Dispose()
    {
        _instance.Clear();
        DefaultDelegatePool<TPooled>.Instance.Free(_instance);
    }
}
```

Reply:
It seems likely. The complexity was necessary in an earlier iteration.

Reply:
> `struct Releaser<TPooled>`

I prefer to avoid this because it forces all the nested helper types to be exposed.

Reply:
IClearable is now removed.